File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(b):
    dic = {"aya":"1","ye":"2","woo":"3","ma":"4"}
    ans = 0
    arr = []
    for word in b:
        for k in dic.keys():
            word = word.replace(k,dic[k])
        arr.append(word)
    for word in arr:
        char = '.'
        flag = True
        for c in word:
            if not c.isdigit() or c == char:
                flag = False
                break
            char = c
        if flag: ans += 1
    return ans

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.kt
class Solution {
    fun solution(bab: Array<String>): Int {
        var ans = 0
        // 연속발음할 순 없다.
        for (i in bab.indices) {
            bab[i] = bab[i]
             .replace("aya","2")
             .replace("ye","3")
             .replace("woo","4")
             .replace("ma","1")
        }
        for (b in bab) {
            var flag = true
            var prev = '.'
            for (chr in b) {
                if (!chr.isDigit() || chr == prev) {
                    flag = false
                    break
                }
                prev = chr
            }
            if (flag) ans ++
        }
        return ans
    }
}